### PR TITLE
docs: disable prettier for docs md

### LINKED ISSUE #216: prettier breaks markdown in the docs
Markdown relies on specific line breaks and indentation, especially for nesting list items

As it is now, prettier is required for each docs PR, and when it's run, it undoes markdown indentation, breaking lists and making a pretty difficult experience

Check out this commit: https://github.com/coder/coder/pull/15527/commits/63c5136350af58743a53f4e7a176eef21e263593

The previous commit failed CI, but this one passes because `make fmt` was run, but in doing so, the list items are a mess:

Image

There are markdown linters and github actions that can enforce style guides. Is it worth adding the docs directory to [.prettierignore.include](https://github.com/coder/coder/blob/main/.prettierignore.include) ?

### PR DESCRIPTION
**Pull request by EdwardAngert**
closes https://github.com/coder/internal/issues/216

There are other md linters that we can use to standardize md rules and formatting, but the current implementation of prettier breaks more than it improves

[preview of jfrog-artifactory](https://coder.com/docs/@216-disable-prettier-docs-md/admin/integrations/jfrog-artifactory)

- [ ] fix existing errors
- [ ] enable more [rules](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md)
- [ ] exclude docs that are generated, such as CLI and API reference
- [ ] bonus - have mdlinter only run on PRs that touch files in docs (@matifali - any ideas on how we could implement this?)

### CODE REVIEW

**Review by bpmct — COMMENTED** [review]
> There are other md linters that we can use to standardize md rules and formatting, but the current implementation of prettier breaks more than it improves

Is there anything we can do to ensure our markdown formats remain consistent in the short-term?

**Review by bpmct — APPROVED** [review]

**Review by matifali — APPROVED** [review]
Could we add something like [markdownlint](https://github.com/DavidAnson/markdownlint-cli2) as the replacement?

**Comment by EdwardAngert** [comment]
> Could we add something like [markdownlint](https://github.com/DavidAnson/markdownlint-cli2) as the replacement?

@matifali , I think so? I feel like they should work together - https://github.com/DavidAnson/markdownlint/blob/main/doc/Prettier.md

**Comment by EdwardAngert** [comment]
@matifali @bpmct did some [testing on my own fork](https://github.com/coder/coder/actions/runs/12187664797/job/33998995120#step:15:18) and it looks like we can get md-linting into the CI without much trouble. The way I've got it now, it'll report errors and won't autofix, but I think that's okay for now

I'll push the changes to this branch

**Comment by EdwardAngert** [comment]
seems to work - [fails successfully](https://github.com/coder/coder/actions/runs/12188699902/job/34002310304?pr=15654#step:15:16)

(now we just have 65 or so issues to fix)

File: .prettierignore.include [hunk 1]
```diff
@@ -12,3 +12,6 @@ testdata/
 # Everything in site/ is formatted by Biome. For the rest of the repo though, we
 # need broader language support.
 site/
+
+# Prettier messes up markdown alignment in the documentation
+**/docs/*.md
```
Comment (matifali) [inline .prettierignore.include:17]: is it needed here?

File: .prettierignore [hunk 2]
```diff
@@ -94,3 +94,6 @@ testdata/
 # Everything in site/ is formatted by Biome. For the rest of the repo though, we
 # need broader language support.
 site/
+
+# Prettier messes up markdown alignment in the documentation
+**/docs/*.md
```
Comment (matifali) [inline .prettierignore:99]:
```suggestion
/docs/*.md
```
This should work too.

File: docs/admin/integrations/jfrog-artifactory.md [hunk 3]
```diff
@@ -31,7 +31,7 @@ by using our official Coder [modules](https://registry.coder.com). We publish
 two type of modules that automate the JFrog Artifactory and Coder integration.
 
 1. [JFrog-OAuth](https://registry.coder.com/modules/jfrog-oauth)
-2. [JFrog-Token](https://registry.coder.com/modules/jfrog-token)
+1. [JFrog-Token](https://registry.coder.com/modules/jfrog-token)
```
Comment (matifali) [inline docs/admin/integrations/jfrog-artifactory.md:34]: Noticed this repetition of indices.

**Review by matifali — COMMENTED** [review]
I left a few suggestions.

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -221,6 +221,13 @@ jobs:
           ./scripts/check_unstaged.sh
         shell: bash
 
+      - name: markdownlint-cli2-action
+        uses: DavidAnson/markdownlint-cli2-action@v18.0.0
+        with:
+          config: ".github/actions/markdownlint-cli2/.markdownlint.json"
+          globs: "docs/**/*.md"
+          fix: true
```
[the remark this answers is not in the document]
Comment (EdwardAngert) [reply .github/workflows/ci.yaml:229]: good call - I was trying to find a way for it to PR with suggested changes/fixes. Pretty sure [megalinter can be configured to](https://megalinter.io/latest/config-apply-fixes/) but that doesn't have to happen now

File: .prettierignore.include (hunk 1 above)
Comment (EdwardAngert) [reply .prettierignore.include:17]: yeah - goal for now is to keep prettier from messing up the markdown spacing

**Comment by matifali** [comment]
> have mdlinter only run on PRs that touch files in docs

I think this is possible with a workflow setting like this. https://github.com/DavidAnson/markdownlint-cli2-action/blob/main/.github/workflows/changed.yml

**Comment by EdwardAngert** [comment]
@matifali !!! It worked! Once it's active, it'll check all new pushes and PRs - check out ,https://github.com/coder/coder/actions/runs/12245139564/job/34158398057?pr=15806#step:4:11>

**Comment by EdwardAngert** [comment]
linter now ignores generated docs in docs/reference

[before](https://github.com/coder/coder/actions/runs/12245097177/job/34158260735?pr=15654) (fails) | [after](https://github.com/coder/coder/actions/runs/12263255808/job/34214383191) (passes)

**Comment by EdwardAngert** [comment]
closing in favor of https://github.com/coder/coder/pull/15831

**Pull request closed without merging**